app/models/host_aspect_base.rb

```diff
@@ -0,0 +1,36 @@
+class HostAspectBase < ActiveRecord::Base
+  self.abstract_class = true
+
+  def self.inherited(subclass)
+    super
+    subclass.class_eval do
+      belongs_to_host
+    end
+  end
+
+  def info
+    {}
+  end
+
+  def smart_proxy_ids
+    []
+  end
+
+  def after_clone
+  end
+
+  def template_filter_options(kind)
+    {}
+  end
+
+  def provisioning_template_options(base_options)
+    base_options
+  end
+
+  def self.inherited_attributes(hostgroup, aspect_attributes)
+    aspect_attributes
+  end
+
+  def self.populate_fields_from_facts(host, importer, type, proxy_id)
```

Comment:
The definition on `import_host_and_facts` on `Host::Managed` doesn't *need* a proxy, it's nil if it can't be found. Otherwise you couldn't submit facts from non-registered proxies, or from [foreman_ansible](https://github.com/dlobatog/foreman_ansible)